dissect/target/plugins/os/default/network.py

```diff
@@ -31,15 +34,16 @@ def check_compatible(self) -> None:
     def _interfaces(self) -> Iterator[InterfaceRecord]:
         yield from ()
 
-    def _get_record_type(self, field_name: str) -> Iterator[Any]:
+    def _get_record_type(self, field_name: str, func: Callable[[Any], Any] | None = None) -> Iterator[Any]:
+        func = func or (lambda _: _)
         for record in self.interfaces():
             if (output := getattr(record, field_name, None)) is None:
                 continue
 
-            if isinstance(output, list):
-                yield from output
-            else:
-                yield output
+            if not isinstance(output, list):
+                output = [output]
+
+            yield from map(func, output)
```

Comment:
```suggestion
            yield from (map(func, output) if func else output)
```
Wouldn't this work? Gets rid of the useless `map` if there's no `func`.